custom_components/ltss/__init__.py

```diff
@@ -274,39 +274,59 @@ def _setup_connection(self):
             self.engine.dispose()
 
         self.engine = create_engine(self.db_url, echo=False,
-            json_serializer=lambda obj: json.dumps(obj, cls=JSONEncoder))
+                                    json_serializer=lambda obj: json.dumps(obj, cls=JSONEncoder))
+
+        inspector = inspect(self.engine)
 
-        # Make sure TimescaleDB  and PostGIS extensions are loaded
         with self.engine.connect() as con:
-            con.execute(
-                text("CREATE EXTENSION IF NOT EXISTS postgis CASCADE"
-                ).execution_options(autocommit=True))
-            con.execute(
-                text("CREATE EXTENSION IF NOT EXISTS timescaledb CASCADE"
-                ).execution_options(autocommit=True))
+            available_extensions = [row['name'] for row in
+                                    con.execute(text("SELECT name FROM pg_available_extensions"))]
+
+            # create table if necessary
+            if not inspector.has_table(LTSS.__tablename__):
+                self._create_table(available_extensions)
+
+            # set/update chunk_time_interval
+            if 'timescaledb' in available_extensions and 1 == con.execute(
+                    f"SELECT 1 FROM timescaledb_information.hypertables "
+                    f"WHERE hypertable_name = '{LTSS.__tablename__}'").rowcount:
```

Comment:
This query seems to be somewhat problematic... It returns with an error on my current setup (timescaledb version 1.4.2, postgis version 2.5.2), yes, I know, slightly outdated. But at this point I am not willing to break backward compatibility in a manner which requires an upgrade of the database. 

For this version of timescaledb, the correct way of checking if a table is a hypertable would be:
```python
f"SELECT 1 FROM _timescaledb_catalog.hypertable WHERE table_name = '{LTSS.__tablename__}"
```

Not sure though how we can guarantee this and the other query covers all versions of timescaledb. Any ideas?